variables.tf

```
@@ -4,30 +4,135 @@
+}
+
+variable "file_header" {
+  default     = "###\n# https://{domain}/{filename}\n# {timestamp}\n# {description}\n###\n"
+  description = "The header template for each published file.  The following variables are available within the template: {domain} - the domain where the published files are located, {filename} - the name of the published file, {timestamp} - the timestamp when the file was published, {description} - the description of the published file"
+  type        = string
+}
+
+variable "lambda_function_description" {
+  default     = "Lambda function to publish egress IP addresses to an S3 bucket configured with a CloudFront distribution for HTTPS access."
+  description = "The description of the Lambda function."
+  type        = string
+}
+
+variable "lambda_function_name" {
+  default     = "publish-egress-ip"
+  description = "The name of the Lambda function to publish egress IP addresses."
+  type        = string
+}
+
+variable "lambda_schedule_interval" {
+  default     = 60
+  description = "The number of minutes between scheduled runs of the Lambda function to publish egress IP addresses.  This value must be an integer greater than 0."
+  type        = number
+
+  validation {
+    condition     = try(parseint(tostring(var.lambda_schedule_interval), 10), null) == var.lambda_schedule_interval && var.lambda_schedule_interval > 0
+    error_message = "lambda_schedule_interval must be an integer greater than zero."
+  }
+}
+
+variable "lambda_zip_filename" {
+  default     = "publish_egress_ip.zip"
+  description = "The name of the ZIP file containing the Lambda function deployment package to publish egress IP addresses.  The file must be located in the root directory of this project."
+  type        = string
+}
+
+variable "lambdaexecution_role_description" {
+  default     = "Allows the publish-egress-ip Lambda to query other accounts for public EC2 IP information, publish objects to the S3 bucket, and write CloudWatch logs."
+  description = "The description to associate with the IAM role (and policy) that allows the publish-egress-ip Lambda to query other accounts for public EC2 IP information, publish objects to the S3 bucket, and write CloudWatch logs."
+  type        = string
+}
+
+variable "lambdaexecution_role_name" {
+  default     = "PublishEgressIPLambda"
+  description = "The name to assign the IAM role (and policy) that allows the publish-egress-ip Lambda to query other accounts for public EC2 IP information, publish objects to the S3 bucket, and write CloudWatch logs."
+  type        = string
+}
+
+variable "publish_egress_tag" {
+  default     = "Publish Egress"
+  description = "The name of the AWS tag whose value represents whether the EC2 instance or elastic IP should have its public IP address published."
```

Mirroring the suggestion in `README.md`.

```suggestion
  description = "The name of the AWS resource tag whose value represents whether the EC2 instance or elastic IP should have its public IP address published."
```